### ISSUE TITLE
Character Seq2Seq Attention model

### ISSUE THREAD

**Issue by shubhamagarwal92** [google/seq2seq#113]
For training character level model, do we have to provide the delimiter for training as  
python -m bin.train \
.....
--input_pipeline_train "
class: ParallelTextInputPipeline
    params:
      source_delimiter :
        -""
      target_delimiter :
       -""
.....
"
Similarly for input_pipeline_dev and also when inferring? 

I tried as suggested in https://google.github.io/seq2seq/tools/,  passing delimiter flag during training.
python -m bin.train \
--delimiter = ""
--config_paths = "....."

but I don't think it is using this parameter (looking at the code as well as the results.) 

For character level seq2seq model, is it necessary to generate vocabulary using generate_vocab.py as solved by #108 or can we use our own defined vocabulary.

**Comment by dennybritz** [comment]
Yes, the delimiter is a parameter of the input pipeline so you need to pass it as part of the train and dev input pipeline:

```
class: ParallelTextInputPipeline
  params:
    source_delimiter: ""
    target_delimiter: ""
```

You may also need to change the delimiter for eval, which is used to join the predicted tokens. Check out the `example_configs/text_metrics.bpe` file and change the `separator` in there.

> I tried as suggested in https://google.github.io/seq2seq/tools/, passing delimiter flag during training.

Thanks for catching, that sentence is outdated.

> For character level seq2seq model, is it necessary to generate vocabulary using generate_vocab.py as solved by #108 or can we use our own defined vocabulary.

You can define your own vocab as long as its in the right format (one token/character per line).

**Comment by sheerun** [comment]
> Thanks for catching, that sentence is outdated and should be removed.

It's nice to open an issue when discovering something like this, so it's noted and fixed at some point
